docs/datastorage/capabilities.md

```diff
@@ -0,0 +1,176 @@
+The Capability System
+==========================
+
+Capabilities allow exposing features in a dynamic and flexible way, without having to resort to directly implementing many interfaces.
+
+In general terms, each capability provides a feature in the form of a class or interface, alongside with a default implementation which can be requested, and a storage handler for at least this default implementation. The storage handler can support other implementations, but this is up to the capability implementor, so look it up in their documentation before trying to use the default storage with non-default implementations.
+
+Forge adds capability support to TileEntities, Entities, and ItemStacks, which can be exposed either by attaching them through an event or, with the exception of ItemStacks, by overriding he capability methods in your own implementations of the entities. This will be explained in more detail in the following sections.
```

Comment:
"overriding the", and Items can overide their own methods to implement the ItemStack one. It's just weird for them because of the instance vs singleton thing. The PREFERRED method is to implement the functions on your own TEs/Entites/Items because you have the code and you can dictate what goes first. The event is just a hacky compromise that we did to appease certain modders.